lib/luminork-server/src/service/v1/components/mod.rs

```diff
@@ -612,6 +627,51 @@ impl ComponentViewV1 {
 
         let attributes: AttributeSources = Component::sources(ctx, component_id).await?.into();
 
+        // Determine if this component is a secret-defining component and get its secret_id
+        // Secret-defining components store the encrypted secret key as the value in their
+        // /root/secrets/<output_socket_name> attribute. We need to reverse-lookup the secret
+        // by that encrypted key value.
+        let secret_id = if SchemaVariant::is_secret_defining(ctx, schema_variant.id()).await? {
+            let output_socket =
+                SchemaVariant::find_output_socket_for_secret_defining_id(ctx, schema_variant.id())
```

Comment:
We should really schedule work to finish off output sockets once and for all. Including the asset builder.